test/extended/router/gatewayapi.go

```diff
@@ -0,0 +1,147 @@
+	)
+
+	g.Describe("Verify Gateway API CRDs", func() {
+		g.It("and ensure required CRDs should already be installed", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Get and check the installed CRDs")
+			for i := range crdNames {
+				crd, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), crdNames[i], metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				e2e.Logf("Found the Gateway API CRD named: %v", crd.Name)
+			}
+		})
+
+		g.It("and ensure existing CRDs can not be deleted", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Try to delete the CRDs and fail")
+			for i := range crdNames {
+				err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Delete(context.Background(), crdNames[i], metav1.DeleteOptions{})
+				o.Expect(err).To(o.HaveOccurred())
+				o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
+			}
+		})
+
+		g.It("and ensure existing CRDs can not be updated", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Get the CRDs firstly, add spec.names.shortNames then update CRD")
+			for i := range crdNames {
+				crd, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), crdNames[i], metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				// some CRDs have a shortName but some not, just trying to add one for all
+				crd.Spec.Names.ShortNames = append(crd.Spec.Names.ShortNames, "fakename")
+				_, err = crdClient.ApiextensionsV1().CustomResourceDefinitions().Update(context.Background(), crd, metav1.UpdateOptions{})
+				o.Expect(err).To(o.HaveOccurred())
+				o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
+			}
+		})
+
+		g.It("and ensure CRD of standard group can not be created", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			fakeCRDName := "fakeroutes.gateway.networking.k8s.io"
+			g.By("Try to create CRD of standard group and fail")
+			fakeCRD := buildGWAPICRDFromName(fakeCRDName)
+			_, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Create(context.Background(), fakeCRD, metav1.CreateOptions{})
+			o.Expect(err).To(o.HaveOccurred())
+			o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
+		})
+
+		g.It("and ensure CRD of experimental group can not be created", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			expCRDName := "xlistenersets.gateway.networking.x-k8s.io"
+
+			g.By("Ensure no CRD of experimental group is installed")
+			crdList, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().List(context.Background(), metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			for _, crd := range crdList.Items {
+				if crd.Spec.Group == "gateway.networking.x-k8s.io" {
+					e2e.Failf("Found unexpected CRD named: %v", crd.Name)
+				}
+			}
```

Comment:
Should this be a separate `It` clause similar to `g.It("and ensure required CRDs should already be installed"`?
This way we won't need 2 `By` clauses.

Reply:
Ah yes, make sense, that way looks neat and focused.